libvast/builtins/formats/json.cpp

```diff
@@ -555,7 +555,9 @@ class default_parser : public parser_base<FieldValidator> {
         this->ctrl_.warn(caf::make_error(
           ec::parse_error, fmt::format("skips invalid JSON '{}' : {}", view,
```

Comment:
The error message is incorrect now

Reply:
I think it was incorrect to begin with.
I have an issue for it: https://github.com/tenzir/issues/issues/638

Reply:
Now that we abort instead of warning it is unlikely to be a wrong error message.
I still think that when we detect a schema change and yield slice just before aborting we may give a wrong error message.

E.g we might have this data in a single chunk
```json
{"pcap_cnt":83,"event_type":"alert"}
{"pcap_cnt":83,"event_type":"dns"}
{"pcap_cnt":83,"event_type":"dns"{}
```
We parse the first event successfully.
We yield the first event because the selector field for the second event changed.
We parse the second event.
We abort on third event.

I think here it might be non deterministic. Will we process the yielded slice first? Or will the call to abort stop the execution.

Notice that it happens only with read json with --selector option. It doesn't happen for zeek/suricata.
I will update the mentionted issue accordingly

Reply:
Let's tackle this in the follow-up issue.